### PR TITLE
(WIP) Triple classification

### PR DESCRIPTION
**Pull request by samuelbroscheit**
I didn't want this PR go to waste so I fixed it up to its current state. It works now except for the  following, that would have to be adressed:

There is an incomplete attempt from me to integrate the TC datasets (where valid and test have a label) WN11 and FB13 into our dataset framework. The question here is, how we want to proceed here. My plan was to hae a `load_labels()` function similar to `load triples()`.

### CODE REVIEW

File: examples/toy-complex-train-tripleclass.yaml [hunk 1]
```diff
@@ -11,8 +11,11 @@ lookup_embedder.dim: 100
 lookup_embedder.initialize: xavier_uniform_
 eval:
   type: triple_classification
-  metrics_per.relation: False
-  triple_classification_random_seed: False
+triple_classification.random_seed: False
```
Comment (samuelbroscheit) [inline examples/toy-complex-train-tripleclass.yaml:14]: triple_classification.random_seed is not used anymore

File: kge/config-default.yaml [hunk 2]
```diff
@@ -423,6 +423,14 @@ valid:
 
 ## EVALUATION ##################################################################
 
+triple_classification:
+  random_seed: False
```
Comment (samuelbroscheit) [inline kge/config-default.yaml:427]: same here: random_seed can be removed

File: kge/job/triple_classification.py [hunk 3]
```diff
@@ -31,6 +36,7 @@ def __init__(self, config: Config, configuration_key: str, dataset: Dataset):
 
     def _prepare(self,):
         train_data = self.dataset.split("train")
+        #TODO probably outdated as it refers to out-commented code
```
Comment (samuelbroscheit) [inline kge/job/triple_classification.py:39]: correct

File: kge/job/triple_classification.py [hunk 4]
```diff
@@ -20,7 +20,12 @@ def __init__(self, config: Config, configuration_key: str, dataset: Dataset):
         self.o_entities = None
         uni_sampler_config = config.clone()
         # uni_sampler_config.set("negative_sampling.num_samples.s", self.get_option("num_samples.s"))
+        # TODO this is redundant as uniform.sample() is called with "num_samples" here in self.sample()
```
Comment (samuelbroscheit) [inline kge/job/triple_classification.py:23]: yes that is correct

File: kge/job/triple_classification.py [hunk 5]
```diff
@@ -20,7 +20,12 @@ def __init__(self, config: Config, configuration_key: str, dataset: Dataset):
...
         uni_sampler_config.set("negative_sampling.num_samples.s", 1)
+        # TODO maybe changing the API of KGEsampler.sample() to also accept a param "filter"
+        #  as it is the case already with "num_samples"
+        #  then we would not rely here on configuration options which actually
+        #  belong to a training job
```
Comment (samuelbroscheit) [inline kge/job/triple_classification.py:28]: Absolutely. Currently the negative sampling config is mixing the Job and the Sampler. This should be split up in a separate PR.

File: kge/job/triple_classification.py [hunk 6]
```diff
@@ -121,7 +141,8 @@ def _prepare(self):
 
         self.config.log("Generate data with corrupted and true triples...")
 
-        if self.eval_split == "test":
+        # TODO maybe should be generalized to allow for other splits as valid_wo_unseen
```
Comment (samuelbroscheit) [inline kge/job/triple_classification.py:144]: Yes I agree, see my comment https://github.com/uma-pi1/kge/pull/115#issuecomment-647194593

File: kge/job/triple_classification.py [hunk 7]
```diff
@@ -130,7 +151,25 @@ def _prepare(self):
...
+            positives_valid = self.dataset.split("valid")
+            negatives_valid = self.dataset.split("valid_negatives")
+            self.tune_data = torch.cat((positives_valid, negatives_valid)).to(self.device)
+            self.tune_labels = torch.cat(
+                (torch.ones(positives_valid.size(0), torch.zeros(negatives_valid.size(0))))
+            ).to(self.device)
+
+            positives_test = self.dataset.split("test")
+            negatives_test = self.dataset.split("test_negatives")
+            self.tune_data = torch.cat((positives_test, negatives_test)).to(
```
Comment (samuelbroscheit) [inline kge/job/triple_classification.py:165]: eval_data

File: kge/job/triple_classification.py [hunk 8]
```diff
@@ -130,7 +151,25 @@ def _prepare(self):
...
+                self.device)
+            self.tune_labels = torch.cat(
```
Comment (samuelbroscheit) [inline kge/job/triple_classification.py:167]: eval_labels
